components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -0,0 +1,399 @@
+        }
+
+        return new ActionExecutionStatus(ActionExecutionStatus.Status.FAILURE, eventContext);
+    }
+
+    private ActionExecutionStatus processSuccessResponse(ActionInvocationSuccessResponse successResponse,
+                                                         ActionType actionType,
+                                                         Map<String, Object> eventContext,
+                                                         ActionExecutionRequest actionRequest,
+                                                         String actionId, String apiEndpoint,
+                                                         AuthMethods.AuthMethod authenticationMethod,
+                                                         ActionExecutionResponseProcessor
+                                                                 actionExecutionResponseProcessor)
+            throws ActionExecutionResponseProcessorException {
+
+        if (LOG.isDebugEnabled()) {
+            // todo: add to diagnostic logs
+            logSuccessResponse(successResponse, actionType, actionId, apiEndpoint, authenticationMethod);
+        }
+
+        List<PerformableOperation> allowedPerformableOperations =
+                validatePerformableOperations(actionRequest, successResponse);
+        ActionInvocationSuccessResponse.Builder successResponseBuilder =
+                new ActionInvocationSuccessResponse.Builder().operations(allowedPerformableOperations);
+        return actionExecutionResponseProcessor.processSuccessResponse(actionType, eventContext,
+                actionRequest.getEvent(),
+                successResponseBuilder.build());
+    }
+
+    private ActionExecutionStatus processErrorResponse(ActionInvocationErrorResponse errorResponse,
+                                                       ActionType actionType,
+                                                       Map<String, Object> eventContext,
+                                                       ActionExecutionRequest actionRequest,
+                                                       String actionId, String apiEndpoint,
+                                                       AuthMethods.AuthMethod authenticationMethod,
+                                                       ActionExecutionResponseProcessor
+                                                               actionExecutionResponseProcessor)
+            throws ActionExecutionResponseProcessorException {
+
+        if (LOG.isDebugEnabled()) {
+            // todo: add to diagnostic logs
+            logErrorResponse(errorResponse, actionType, actionId, apiEndpoint, authenticationMethod);
+        }
+
+        return actionExecutionResponseProcessor.processErrorResponse(actionType, eventContext, actionRequest.getEvent(),
+                errorResponse);
+    }
+
+    private void logSuccessResponse(ActionInvocationSuccessResponse successResponse, ActionType actionType,
```

Comment:
I think the logSuccessResponse, logErrorResponse, and logErrorResponse methods should be moved to a LogUtility class dedicated to Actions/Events. What do you think?

Reply:
That an improvement to consider. Yet let's improve as we onboard more actions.